### PR TITLE
Add extension to Calendar for number of days in year and number of months in year

### PR DESCRIPTION
**Pull request by kaphacius**
2 extensions, apparently days in year and months in year can vary a lot (especially in chinese and hebrew calendars)

## Checklist
- [x] I checked the [**Contributing Guidelines**](https://github.com/SwifterSwift/SwifterSwift/blob/master/CONTRIBUTING.md) before creating this request.
- [x] New extensions are written in Swift 4.
- [x] New extensions support iOS 8.0+ / tvOS 9.0+ / macOS 10.10+ / watchOS 2.0+.
- [x] I have added tests for new extensions, and they passed.
- [x] All extensions have a **clear** comments explaining their functionality, all parameters and return type in English.
- [x] All extensions are declared as **public**.

### CODE REVIEW

File: Sources/Extensions/Foundation/CalendarExtensions.swift [hunk 1]
```diff
@@ -20,4 +20,23 @@ public extension Calendar {
...
+    }
+    
+    /// SwifterSwift: Return the number of months in a year for specified 'date'
+    ///
+    ///         let calendar = Calendar(identifier: .gregorian)
+    ///         calendar.numberOfMonthsInYear(for: Date()) -> 12
+    ///
+    /// - Parameter date: the date from which the year is obtained
+    /// - Returns: number of days in the year of 'date'
+    public func numberOfMonthsInYear(for date: Date) -> Int {
```
Comment (SD10) [inline Sources/Extensions/Foundation/CalendarExtensions.swift]: Are there other calendars with more months than 12? I'm not sure if this is necessary

File: Sources/Extensions/Foundation/CalendarExtensions.swift [hunk 2]
```diff
@@ -20,4 +20,23 @@ public extension Calendar {
     public func numberOfDaysInMonth(for date: Date) -> Int {
         return range(of: .day, in: .month, for: date)?.count ?? 0
     }
+    
+    /// SwifterSwift: Return the number of days in the year for specified 'Date'.
+    ///
+    /// - Parameter date: the date from which the year is obtained
+    /// - Returns: number of days in the year of 'date'
+    public func numberOfDaysInYear(for date: Date) -> Int {
```
Comment (SD10) [inline Sources/Extensions/Foundation/CalendarExtensions.swift]: I'd say the same about this one... but I guess some years can have 366 days with leap year

File: Tests/FoundationTests/CalendarExtensionTest.swift [hunk 3]
```diff
@@ -28,5 +28,34 @@ class CalendarExtensionTests: XCTestCase {
...
-	
+    
+    func testNumberOfMonthsInYear() {
+        let gregorian = Calendar(identifier: .gregorian)
+        let hebrew = Calendar(identifier: .hebrew)
+        let ethiopic = Calendar(identifier: .ethiopicAmeteMihret)
+        let today = Date()
+        
+        XCTAssert(gregorian.numberOfMonthsInYear(for: today) == 12)
+        XCTAssert(hebrew.numberOfMonthsInYear(for: today) == 13)
```
Comment (SD10) [inline Tests/FoundationTests/CalendarExtensionTest.swift]: Should've looked at the tests first :joy:

Comment (kaphacius) [reply Tests/FoundationTests/CalendarExtensionTest.swift]: what do you mean?

**Comment by kaphacius** [comment]
@SD10 done

File: CHANGELOG.md [hunk 4]
```diff
@@ -32,6 +32,10 @@ All notable changes to this project will be documented in this file.
    -    Add `scaled(toMaxSize:)` to scale image to maximum size with respect to aspect ratio [#291](https://github.com/SwifterSwift/SwifterSwift/pull/291) by [buddax2](https://github.com/buddax2).
  - **Optional**
    - Add optional assignment operator `??=` [#296](https://github.com/SwifterSwift/SwifterSwift/pull/296) by [buddax2](https://github.com/buddax2).
+ - **Calendar**
+   - Add `numberOfDaysInMonth` [#311](https://github.com/SwifterSwift/SwifterSwift/pull/311) by [chaithanyaprathyush](https://github.com/chaithanyaprathyush).
+   - Add `numberOfMonthsInYear` [#315](https://github.com/SwifterSwift/SwifterSwift/pull/315) by [kaphacius](https://github.com/kaphacius).
+   - Add `numberOfDaysInYear` [#315](https://github.com/SwifterSwift/SwifterSwift/pull/315) by [kaphacius](https://github.com/kaphacius).
```
Comment (SD10) [inline CHANGELOG.md]: Next time referencing the PR once will do just fine 😅

**Comment by kaphacius** [comment]
@SD10 yes ofc, apparently it works differently on osx and ios, i have to rewrite the extension

**Comment by SD10** [comment]
@kaphacius Any update on this one?

**Comment by kaphacius** [comment]
@SD10 sorry, i'll work on this during the weekend

**Comment by SD10** [comment]
@kaphacius No rush, thanks 👍

**Comment by omaralbeik** [comment]
@kaphacius I'd close this for now, feel free to open it again when ready :)

**Pull request closed without merging**